helios-services/src/test/java/com/spotify/helios/agent/TaskRunnerTest.java

```diff
@@ -113,4 +129,49 @@ public void testPullTimeoutVariation() throws Throwable {
       assertEquals(ImagePullFailedException.class, t.getCause().getClass());
     }
   }
+
+  @Test
+  public void testContainerNotRunningVariation() throws Throwable {
+
+    final ContainerInfo stopped = new ContainerInfo() {
+      @Override
+      public ContainerState state() {
+        final ContainerState state = mock(ContainerState.class);
+        when(state.running()).thenReturn(false);
+        when(state.error()).thenReturn("container is a potato");
+        return state;
+      }
+    };
+
+    doReturn("potato").when(mockCreation).id();
+    doReturn(stopped).when(mockDocker).inspectContainer(anyString());
+    doReturn(mockImageInfo).when(mockDocker).inspectImage(IMAGE);
+    doReturn(mockCreation).when(mockDocker)
+            .createContainer(any(ContainerConfig.class), anyString());
+    doReturn(false).when(mockHealthChecker).check(anyString());
+
+    final TaskRunner tr = TaskRunner.builder()
+            .delayMillis(0)
+            .config(TaskConfig.builder()
+                    .namespace("test")
+                    .host(HOST)
+                    .job(JOB)
+                    .containerDecorators(ImmutableList.of(containerDecorator))
+                    .build())
+            .docker(mockDocker)
+            .listener(mockListener)
+            .healthChecker(mockHealthChecker)
+            .build();
+
+    tr.run();
+
+    try {
+      tr.resultFuture().get();
+      fail("this should throw");
```

Comment:
Hmm, I guess that might not necessarily work if you use the future, via throwing of the exception, as a synchronization point which it kind of looks like you do? :)